HealthcareSystem/Backend/Repository/DrugConsumptionRepository/MySqlDrugConsumptionRepository/MySqlDrugConsumptionRepository.cs

```diff
@@ -0,0 +1,72 @@
+﻿using Backend.Model;
+using Backend.Model.Pharmacies;
+using Microsoft.EntityFrameworkCore.Internal;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Backend.Repository.DrugConsumptionRepository.MySqlDrugConsumptionRepository
+{
+    public class MySqlDrugConsumptionRepository : IDrugConsumptionRepository
+    {
+        private readonly MyDbContext _context;
+
+        public MySqlDrugConsumptionRepository(MyDbContext context)
+        {
+            _context = context;
+        }
+        public IEnumerable<DrugConsumption> GetAllDrugConsumptions()
+        {
+            return _context.DrugConsumptions.ToList();
+        }
+
+        public DrugConsumption GetDrugConsumptionById(int id)
+        {
+            return _context.DrugConsumptions.Find(id);
+        }
+
+        public void CreateDrugConsumption(DrugConsumption dc)
+        {
+            _context.Add(dc);
+            _context.SaveChanges();
+        }
+
+        public void UpdateDrugConsumption(DrugConsumption dc)
+        {
+            _context.Update(dc);
+            _context.SaveChanges();
+        }
+
+        public void DeleteDrugConsumption(DrugConsumption dc)
+        {
+            _context.Remove(dc);
+            _context.SaveChanges();
+        }
+
+        public IEnumerable<DrugConsumptionReport> GetDrugConsumptionForDate(DateRange dateRange)
+        {
+            var query = _context.DrugConsumptions
+                .Where(d => d.Date >= dateRange.From)
+                .Where(d => d.Date <= dateRange.To)
+                .Join(
+                    _context.Drugs,
+                    dc => dc.DrugId,
+                    d => d.Id,
+                    (dc, d) => new
+                    {
+                        DrugId = d.Id,
+                        DrugName = d.Name,
+                        TotalConsumption = dc.Quantity
+                    })
```

Comment:
I'm not the most familiar with Entity Framework, but I believe it might be possible to avoid manually writing a join here if you add a reference to Drug in consumption along with the id? It would probably do an implicit join via the lazy loading proxy when you accessed that field then. I'm not 100% what EF will do implicitly and what not though, I definitely don't mind how you implemented things here. I really like that you're doing this nicely with the group by.

Reply:
Honestly, I am not that well versed in EF either. I actually started by writing an SQL query for what I needed and then turned it into Linq syntax. If someone who's more experienced with EF knows how to make this better, they are more than welcome to, but I think it's fine as it is for now.